### PR TITLE
[Enhancement] Using separate memtable flush executor for shared-data mode

### PR DESCRIPTION
**Pull request by xiangguangyxg**
## Why I'm doing:
The flush_thread_num_per_store config item is not suitable in shared-data mode.

## What I'm doing:
Add  lake_flush_thread_num_per_store config item and use a separate memtable flush executor for shared-data mode.

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [x] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [x] This pr needs user documentation (for new or modified features or behaviors)
  - [x] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/http/action/update_config_action.cpp [hunk 1]
```diff
@@ -130,6 +130,14 @@ Status UpdateConfigAction::update_config(const std::string& name, const std::str
             (void)StorageEngine::instance()->segment_flush_executor()->update_max_threads(
                     config::flush_thread_num_per_store * dir_cnt);
         });
+        _config_callback.emplace("lake_flush_thread_num_per_store", [&]() {
+            int threads = config::lake_flush_thread_num_per_store;
+            if (threads <= 0) {
+                threads = CpuInfo::num_cores() * 2;
```
Comment (kevincai) [inline be/src/http/action/update_config_action.cpp:136]: can be extended to if `<=0`, means liner to the the num_cores. e.g. -2 means 2*cpu_cores, -3 means 3 * cpu_cores.

**Review by kevincai — DISMISSED** [review]

**Review by sduzh — DISMISSED** [review]

**Review by sduzh — APPROVED** [review]

File: be/src/http/action/update_config_action.cpp (hunk 1 above)
Comment (xiangguangyxg) [reply be/src/http/action/update_config_action.cpp]: done

**Review by kevincai — APPROVED** [review]

File: docs/zh/administration/management/BE_configuration.md [hunk 2]
```diff
@@ -1803,6 +1803,15 @@ curl http://<BE_IP>:<BE_HTTP_PORT>/varz
 - 描述：每个 Store 用以 Flush MemTable 的线程数。
 - 引入版本：-
 
+##### lake_flush_thread_num_per_store
+
+- 默认值：0
+- 类型：Int
+- 单位：-
+- 是否动态：是
+- 描述：在存算分离模式下，每个 Store 用以 Flush MemTable 的线程数。当该参数被设置为小于或等于 `0` 时，系统默认使用 CPU 核数的两倍
```
Comment (wyb) [inline docs/zh/administration/management/BE_configuration.md:1812]: fix when `lake_flush_thread_num_per_store` < 0

Comment (xiangguangyxg) [reply docs/zh/administration/management/BE_configuration.md:1812]: in other pr

**Pull request merged**